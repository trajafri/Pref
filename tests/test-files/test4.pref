
(define append
  (fix (lambda (append l1 l2)
         (if (empty? l1) l2
           (if (empty? l2) l1
             (cons (car l1) (append (cdr l1) l2)))))))

(append (cons 1 (cons 2 (cons 3 empty)))
        (cons 4 (cons 5 (cons 6 empty))))

(define map
  (fix (lambda (map f ls)
         (if (empty? ls) empty (cons (f (car ls)) (map f (cdr ls)))))))

(define powerset
  (fix (lambda (powerset ls)
         (if (empty? ls) (cons empty empty)
             (append (powerset (cdr ls))
                     (map (cons (car ls)) (powerset (cdr ls))))))))

(powerset (cons 1 (cons 2 (cons 3 empty))))
