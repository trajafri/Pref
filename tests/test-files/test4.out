(cons 1 (cons 2 (cons 3 (cons 4 (cons 5 (cons 6 empty))))))
(cons  empty
  (cons (cons 3 empty) 
    (cons (cons 2 empty) 
      (cons (cons 2 (cons 3 empty))
        (cons (cons 1 empty)
          (cons (cons 1 (cons 3 empty))
            (cons (cons 1 (cons 2 empty)) 
              (cons (cons 1 (cons 2 (cons 3 empty))) empty))))))))
